ksql-rest-app/src/main/java/io/confluent/ksql/rest/client/KsqlRestClient.java

```diff
@@ -90,21 +91,49 @@
   private final LocalProperties localProperties;
 
   public KsqlRestClient(final String serverAddress) {
-    this(serverAddress, Collections.emptyMap());
+    this(serverAddress, Collections.emptyMap(), Collections.emptyMap());
   }
 
-  public KsqlRestClient(final String serverAddress, final Properties properties) {
-    this(serverAddress, propertiesToMap(properties));
+  /**
+   * @param serverAddress the address of the KSQL server to connect to.
+   * @param localProperties initial set of local properties.
+   * @param clientProps properties used to build the client.
+   */
+  public KsqlRestClient(
+      final String serverAddress,
+      final Map<String, ?> localProperties,
+      final Map<String, String> clientProps
```

Comment:
I don't follow...  it's already split.

- `clientProps` is only used to create the client and is not referenced by the client.
- `localProps` is what is kept as a reference and used when making requests to the server.

Reply:
Essentially, I feel that it is important to name anything that contains sensitive information with a name that reflects that fact that it might contain sensitive information. In this case `clientProps` doesn't scream out that it holds a plaintext password, while `sslProps` would. Since today they are one and the same, a simple rename would work, but I imagine we may have more client props in the future that do not contain sensitive information that keeping around would be valuable.